### PR TITLE
boost 1.75

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

## Update

|Formula|CI Failure|Version|Status|
|---|---|---|---|
|cucumber-cpp|test|10.14|Unable to install Ruby Gems, cf https://github.com/Homebrew/homebrew-core/issues/66450|
|boost-python|test|10.15,11.0| Not due to `boost` version bump, see below |

`boost-python` doesn't have a Big Sur bottle (https://github.com/Homebrew/homebrew-core/issues/65000), so I suspect this just isn't compatible with Xcode 12.

I verified this by testing `boost-python` against the existing `boost` formula (pre-version bump) on a GitHub macOS 10.15 runner here: https://github.com/carlocab/homebrew-personal/pull/21/checks?check_run_id=1580808244

The reported error is the same, so the `boost-python` breakage is not due to the `boost` version bump.

## Non-deterministic Failures (CI-induced?)

These failures _sometimes_ emerge, and sometimes don't. I suspect they are `mariadb`-type CI failures (ref https://github.com/Homebrew/homebrew-core/issues/66450).
|Formula|CI Failure|Version|Error Message|
|---|---|---|---|
|cucumber-cpp|`test`|10.15,11.0|`Unable to contact the wire server at localhost:3902.`|
|freeling|`bottle`|11.0|`gzip: write: No space left on device`|
|widelands|`install --build-bottle`|11.0|`Write failed`|

For example, in [this test run](https://github.com/Homebrew/homebrew-core/pull/66792/checks?check_run_id=1575956153) `cucumber-cpp` errored for 11.0 but not for 10.15. It errored out for both versions with the same error in the [previous run](https://github.com/Homebrew/homebrew-core/runs/1565898098). (Neither produced errors on the third-to-the-last run.)

See also `fn` and `frugal` failures at https://github.com/Homebrew/homebrew-core/pull/66355.

### CODE REVIEW

**Comment by carlocab** [comment]
10.14
```
Error: 17 failed steps!
brew test --retry --verbose avro-cpp
brew test --retry --verbose cucumber-cpp
brew test --retry --verbose freeling
brew test --retry --verbose ledger
brew test --retry --verbose lgogdownloader
brew test --retry --verbose libbitcoin-client
brew test --retry --verbose libbitcoin-explorer
brew test --retry --verbose libbitcoin-protocol
brew test --retry --verbose libbitcoin-server
brew test --retry --verbose libpulsar
brew fetch --retry mapnik
brew test --retry --verbose monero
brew test --retry --verbose ncmpcpp
brew test --retry --verbose pcl
brew test --retry --verbose source-highlight
brew test --retry --verbose widelands
brew test --retry --verbose wownero
```

**Comment by carlocab** [comment]
10.15
```
Error: 18 failed steps!
brew test --retry --verbose avro-cpp
brew test --retry --verbose boost-python
brew test --retry --verbose cucumber-cpp
brew test --retry --verbose freeling
brew test --retry --verbose ledger
brew test --retry --verbose lgogdownloader
brew test --retry --verbose libbitcoin-client
brew test --retry --verbose libbitcoin-explorer
brew test --retry --verbose libbitcoin-protocol
brew test --retry --verbose libbitcoin-server
brew test --retry --verbose libpulsar
brew fetch --retry mapnik
brew test --retry --verbose monero
brew test --retry --verbose ncmpcpp
brew test --retry --verbose pcl
brew test --retry --verbose source-highlight
brew test --retry --verbose widelands
brew test --retry --verbose wownero
```

**Comment by carlocab** [comment]
There are a lot of `Symbol not found` errors, so I'm guessing that a bunch of these formulae need to be rebuilt against the new boost.

**Comment by carlocab** [comment]
Did the `mapnik` bottles disappear?
```
==> brew fetch --retry mapnik
==> FAILED
==> Downloading https://homebrew.bintray.com/bottles/mapnik-3.0.23_4.mojave.bottle.tar.gz
curl: (22) The requested URL returned error: 404 Not Found
```

**Comment by carlocab** [comment]
For `pcl`:
```
-- Checking for module 'eigen3'
--   No package 'eigen3' found
-- Could NOT find Eigen (missing: EIGEN_INCLUDE_DIR) (Required is at least version "3.1")
```

Looks like this just needs `pkg-config` at the test.

**Comment by carlocab** [comment]
11.0:
```
Error: 18 failed steps!
brew test --retry --verbose avro-cpp
brew test --retry --verbose boost-python
brew test --retry --verbose cucumber-cpp
brew test --retry --verbose freeling
brew test --retry --verbose ledger
brew test --retry --verbose lgogdownloader
brew test --retry --verbose libbitcoin-client
brew test --retry --verbose libbitcoin-explorer
brew test --retry --verbose libbitcoin-protocol
brew test --retry --verbose libbitcoin-server
brew test --retry --verbose libpulsar
brew fetch --retry mapnik
brew test --retry --verbose monero
brew test --retry --verbose ncmpcpp
brew test --retry --verbose pcl
brew test --retry --verbose source-highlight
brew test --retry --verbose widelands
brew test --retry --verbose wownero
```

**Comment by carlocab** [comment]
10.14:
```
Error: 6 failed steps!
brew test --retry --verbose cucumber-cpp
brew fetch --retry mapnik
brew test --verbose libbitcoin-protocol
brew test --verbose libbitcoin-client
brew test --verbose libbitcoin-explorer
brew test --verbose libbitcoin-server
```

10.15, 11.0:
```
Error: 7 failed steps!
brew test --retry --verbose cucumber-cpp
brew fetch --retry mapnik
brew test --verbose libbitcoin-protocol
brew test --verbose libbitcoin-client
brew test --verbose boost-python
brew test --verbose libbitcoin-explorer
brew test --verbose libbitcoin-server
```

**Comment by carlocab** [comment]
10.14:
```
Error: 2 failed steps!
brew fetch --retry mapnik
brew test --verbose cucumber-cpp
```
Ok, guess I wasn't able to fix `cucumber-cpp`.

**Comment by carlocab** [comment]
This is the boost-python error:
```
==> python
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ImportError: dlopen(./hello.so, 2): no suitable image found.  Did find:
	file system relative paths not allowed in hardened programs
```

File: Formula/boost-python.rb [hunk 1]
```diff
@@ -1,9 +1,9 @@
 class BoostPython < Formula
```
Comment (SMillerDev) [inline Formula/boost-python.rb:1]: Is this formula still needed anywhere? Otherwise it should probably be deprecated given the state of python 2

Comment (carlocab) [reply Formula/boost-python.rb:1]: I was thinking that too, but then got distracted by the other formula. Can we just remove it? I don't know if I can get it to build.

Fine, we probably can't, it has ~30,000 installs in the past year

Comment (carlocab) [reply Formula/boost-python.rb:1]: Oh, as for being needed anywhere: no, no formulae depend on it.

Comment (SMillerDev) [reply Formula/boost-python.rb:1]: It should probably be deprecated from January 1, 2020 then.

**Comment by carlocab** [comment]
Ok, I don't get why `cucumber-cpp` got broken on 10.15 and 11.0 again. It already tested fine on those versions on the last full testing run.

**Comment by carlocab** [comment]
I think this might be ready, @fxcoudert. Test result summary in the first post.

**Comment by carlocab** [comment]
On a separate note, is there a way to specify `depends_on xcode` but requiring a _maximum_ `xcode` version? Because I think that's what `boost-python` needs.

**Comment by fxcoudert** [comment]
Thanks @carlocab, merged!
I think we could do a maximum Xcode version with `fails_with`

**Comment by carlocab** [comment]
Huh, I thought `fails_with` was just for build requirements. Let me look into it.

**Pull request closed without merging**